pkg/storage/azurecdn/storage.go

```diff
@@ -67,10 +68,19 @@ func (s *Storage) Save(ctx context.Context, module, version string, mod, zip, in
 		_, err := url.Upload(ctx, content, httpHeaders(contentType), emptyMeta, emptyBlobAccessCond)
 		uploadErrs <- err
 	}
+	zipRs, ok := zip.(io.ReadSeeker)
```

Comment:
Is there a better way?

Reply:
the point of this task should be to ease up on memory and not store the whole content of the zip file there. which is still happening with buffer somewhere up as well as with ioutil readall here. at least it looks like it - i just woke up

this one is not easy and i imagine you need to stream bytes as they come from storage or client.
so if we consume 20mb of RAM being idle, we should process 20mb of ram processing zips as well. there should be no difference.

Reply:
Thanks for the comments @michalpristas ! 
You're right the module.MakeZip func uses Buffer but I guess I could use a io.Pipe there. What do you think?

About the azure cdn saver here - the problem is, unlike gcp where io.Reader is enough, azure needs an io.ReadSeeker. The func mergedb downloads the zip from another cdn so we could propably stream this directly from one cdn to another but I'm not sure how to do it because of this ReadSeeker requirement...

Reply:
The only thing that comes to my mind is to save the zip to a temp file and upload it from there to the cdn or using something like this https://godoc.org/github.com/aws/aws-sdk-go/service/s3/s3manager

Reply:
yeah file workaround, I thought about that too as a hackiest way how to do this. 
we need to be platform agnostic and provide a generic way which will work for any storage. 
this is not an easy task and would require a lot of playing around, maybe without any clean way how to achieve lower consumption. 

I will think about it as well. I'm just very limited in time these days.

Reply:
Does it have to be generic for all types of storage? Couldn't we leave this to each storage implementation? i.e. azure provides api for uploading streams - `azblob.UploadStreamToBlockBlob` gcp and aws can do this as well.

Reply:
yeah for sure. what i meant is that we need to design funcs in a way that func signatures are the same and all storages can sit behind one interface